File: 242-valid-anagram/valid-anagram.java
class Solution {
    private HashMap<Character,Integer> freqMap(String str){
        HashMap<Character, Integer> map = new HashMap<>();

        for (char ch : str.toCharArray()){
            map.put(ch, map.getOrDefault(ch, 0 ) + 1);
        }

        return map;
    }

    public boolean isAnagram(String s, String t) {
        HashMap<Character,Integer> freqMap1 = freqMap(s);
        HashMap<Character,Integer> freqMap2 = freqMap(t);

        return freqMap1.equals(freqMap2);
    }

}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return self.count_helper(s) == self.count_helper(t)

    def count_helper(self, s):
        count = {}
        
        for char in s:
            if char not in count:
                count[char] = 0

            count[char] += 1

        return count
        